src/Shape.cpp

```diff
@@ -28,74 +49,72 @@ bool Shape::intersect(const Ray& /*r*/, float /*t0*/, float /*t1*/, double& /*di
     return false;
 }
 
-/**
- * Check whether a point p is within distance radius of the line segment whose first vertex is described by v and second vertex by v-e
- **/
-bool check_linesegment(const Vector3& p, const double radius, const Vector3& v, const Vector3& e){
-    double d1 = (v-p).length2();  // distance between v and p, squared
-    double d2 = e.dot(v-p);  // dot product between e and v-p
-    if (d2>0) {
-        d2 = d2*d2 / e.length2(); // distance between v and the projection of p on e
-        return d1-d2 < radius && d2 < e.length2();
-    }
-    return false;
-}
-
 /**
  *  @math http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.49.9172&rep=rep1&type=pdf
```

Comment:
In your commit msgs I saw you used another source. Please cite that source so we can see which conventions are followed.